dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java

```diff
@@ -378,14 +390,54 @@ public void setup() throws Exception {
             LibertyClientFactory.getLibertyClient(clientName);
     }
 
-    private static String getReplacementFeature(String originalFeature, Set<String> featuresToAdd) {
+    /**
+     * Obtain the replacement for a feature for this replacement action.
+     * 
+     * The lookup uses the base feature name, which is the feature name up to
+     * the first '-' character.
+     * 
+     * If EE9 replacement is active, allow the base feature name to be replaced
+     * with an EE9 replacement.
+     * 
+     * If no replacement is located, answer null, which indicates that the feature
+     * should be removed instead of being replaced.
+     *
+     * Feature names are required to have a '-', for example, "servlet-3.1".  Null
+     * is answered for feature names which do not have a '-'.
+     *
+     * @param originalFeature The feature name which is to be replaced.
+     * @param replacementFeatures Table of replacement features.
+     *
+     * @return The replacement feature name.  Null if no replacement is available.
+     */
+    private static String getReplacementFeature(String originalFeature, Set<String> replacementFeatures) {
+        String methodName = "getReplacementFeature";
         // Example: servlet-3.1 --> servlet-4.0
-        String featureBasename = originalFeature.substring(0, originalFeature.indexOf('-') + 1); // "servlet-"
-        for (String featureToAdd : featuresToAdd)
-            if (featureToAdd.contains(featureBasename)) // "servlet-4.0".contains("servlet-")
-                return featureToAdd;
+        int dashOffset = originalFeature.indexOf('-');
+        if ( dashOffset == -1 ) {
+            Log.info(c, methodName, "Remove feature [ " + originalFeature +  " ]: No '-' was found.");
+            return null;
+        }
+        // "servlet-3.1" ==> "servlet"
+        String baseFeature = originalFeature.substring(0, dashOffset);
+        if ( JakartaEE9Action.isActive() ) {
+            String ee9BaseFeature = featuresWithNameChangeOnEE9.get(baseFeature);
+            if ( ee9BaseFeature != null ) {
+                Log.info(c, methodName, "Replace base feature [ " + baseFeature +  " ] with EE9 feature [ " + ee9BaseFeature + " ]");
+                baseFeature = ee9BaseFeature;
+            }
+        }
```

Comment:
Lets move this block after the standard replacement block -- we should prefer standard replacements over the special case replacements generally.

Also, we need to support both directions: `EE7/8 --> EE9` but also `EE9 --> EE7/8`
For this we can simply iterate over `featuresWithNameChangeOnEE9.entrySet()` and check if `replacementFeature` is the key or the value, and if it is, check the opposite entry (value or key respectively) for the replacement.

Reply:
Ok sounds good Ill make the change